### PR TITLE
Use the new ObviousCI and conda-build version baking.

### PR DESCRIPTION
**Pull request by pelson**
First cut at using the new conda interface for version baking. This has required a new version of ObviousCI, so may need a bit of iteration...

### CODE REVIEW

**Comment by pelson** [comment]
Two errors:

On OSX:

```
**Build will continue, but no uploads will take place.**

To automatically upload from this script, define the BINSTAR_TOKEN env variable.

This is done automatically on the travis-ci system once the PR has been merged.

Fetching package metadata: ......

Resolving distributions from 20 recipes... 

Traceback (most recent call last):

  File "/Users/travis/miniconda/bin/obvci_conda_build_dir.py", line 19, in <module>

    Builder.handle_args(args).main()

  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/build_directory.py", line 240, in main

    getattr(self, 'extra_build_conditions', []))

  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/build_directory.py", line 136, in compute_matrix

    cases = vn_matrix.special_case_version_matrix(meta, index)

  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/from_conda_manifest_core_vn_matrix.py", line 147, in special_case_version_matrix

    for numpy_pkg in r.get_pkgs(np_spec):

  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 546, in get_pkgs

    raise NoPackagesFound("No packages found in current %s channels matching: %s" % (config.subdir, ms), [ms.spec])

conda.resolve.NoPackagesFound: No packages found in current osx-64 channels matching: numpy x.x
```

This is because there is no automated build of ObviousCI on OSX.

On  linux64:

```
BUILD START: ecmwf_grib-1.12.1-np110py34_4
Fetching package metadata: ........
Solving package specifications: ..........
The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    jasper-1.900.1             |                3         287 KB  scitools

The following NEW packages will be INSTALLED:

    jasper:      1.900.1-3     scitools
    jpeg:        8d-0          defaults
    libgfortran: 1.0-0         defaults
    numpy:       1.10.1-py34_0 defaults
    openblas:    0.2.14-3      defaults
    openssl:     1.0.2d-0      defaults
    pip:         7.1.2-py34_0  defaults
    python:      3.4.3-2       defaults
    readline:    6.2-2         defaults
    setuptools:  18.4-py34_0   defaults
    sqlite:      3.8.4.1-1     defaults
    tk:          8.5.18-0      defaults
    wheel:       0.26.0-py34_1 defaults
    xz:          5.0.5-0       defaults
    zlib:        1.2.8-0       defaults

Removing old work directory
Source cache directory is: /opt/conda/conda-bld/src_cache
Downloading source to cache: grib_api-1.12.1.tar.gz
Downloading https://software.ecmwf.int/wiki/download/attachments/3473437/grib_api-1.12.1.tar.gz
Success
Extracting download
Applying patch: '/conda-recipes/ecmwf_grib/py3k.patch'
patching file configure.ac
patching file configure
patching file python/gribapi.py
patching file python/gribapi_swig.i
patching file python/numpy.i
patching file python/swig_wrap_array.c
patching file python/swig_wrap_array.py
patching file python/swig_wrap_numpy.c
patching file python/swig_wrap_numpy.py
Package: ecmwf_grib-1.12.1-np110py34_4
source tree in: /opt/conda/conda-bld/work/grib_api-1.12.1
+ /opt/conda/bin/python -c 'import conda_build; print(conda_build.__file__);'
/opt/conda/lib/python3.4/site-packages/conda_build/__init__.py
+ mkdir -p /opt/conda/envs/_build_placehold_placehold_placehold_placehold_placehold_placeho/bin
+ cp /opt/conda/bin/python-config /opt/conda/envs/_build_placehold_placehold_placehold_placehold_placehold_placeho/bin/
cp: cannot stat `/opt/conda/bin/python-config': No such file or directory
```

Looks like this is because there is no python-config with the py34 build that has been produced. We need to disable GRIB building on py3...

File: proj4/meta.yaml [hunk 1]
```diff
@@ -7,10 +7,11 @@ source:
   url: http://download.osgeo.org/proj/proj-4.9.1.tar.gz
 
 build:
+  number: 0  # [osx]
   number: 1  # [not osx]
 
-requirements:
-  run:
+requirements:  # [win]
+  run:  # [win]
```
Comment (ocefpaf) [inline proj4/meta.yaml]: Do we really need all those `# [win]`?

Comment (pelson) [inline proj4/meta.yaml]: Yeah, afraid so. Without them we end up with:

```
requirements:
   run:

about:
   home: ...
```

And conda barfs as it thinks we've passed an empty string to the run requirement. Opportunity for a conda-build PR if you're interested?

Comment (ocefpaf) [inline proj4/meta.yaml]: > And conda barfs as it thinks we've passed an empty string to the run requirement.

Our geos recipes do have an empty string in the run requirement.  Is that a new "feature" of conda-build?

Comment (pelson) [inline proj4/meta.yaml]: Hmmm. Maybe it is an obvious-ci thing then. I'm sure I saw it somewhere...

Comment (ocefpaf) [inline proj4/meta.yaml]: I investigated this a little bit further and latest `obvious-ci`+`conda-build < 1.18.1` is OK with empty fields.
Conda `conda-build==1.18.1` returns an error when `obvious-ci` tries this:

```python
Traceback (most recent call last):
  File "/Users/travis/miniconda/bin/obvci_conda_build_dir.py", line 19, in <module>
    Builder.handle_args(args).main()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/build_directory.py", line 232, in main
    recipe_metas = self.fetch_all_metas()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/build_directory.py", line 206, in fetch_all_metas
    recipe_metas = sort_dependency_order(recipe_metas)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/obvci/conda_tools/build_directory.py", line 85, in sort_dependency_order
    meta.get_value('requirements/build', []))
TypeError: coercing to Unicode: need string or buffer, list found
```

**Pull request merged**